torch/optim/adam.py

```diff
@@ -29,8 +31,8 @@ def __init__(
         params: ParamsT,
         lr: Union[float, Tensor] = 1e-3,
         betas: Tuple[float, float] = (0.9, 0.999),
-        eps: float = 1e-8,
-        weight_decay: float = 0,
+        eps=1e-8,
+        weight_decay=0,
```

Comment:
here too

Reply:
though I see other places where this wasn't necessary--do you know the difference between defining vs not here?

If it's fine to not have a type, I am happy with not having one.

Reply:
Sorry, could you please explain the difference to me? I think mypy interprets weight_decay as an int, so should I specify a float here or change 0 to 0.0?

Reply:
It does look like there's no error and it will allow an int or a float or anything (it might just infer it as Any). From my lil endeavor, it still feels the best to include the type. For example:

```
class Foo:
    def __init__(self, arg, kwarg=0):
        self.arg = arg
        self.kwarg = kwarg

a = Foo(1, "l")

print(5 + a.kwarg)
```
This will not error and will do the type conversion automatically.

But the intention is closer to: 
```
class Foo:
    def __init__(self, arg, kwarg: int = 0):
        self.arg = arg
        self.kwarg = kwarg

a = Foo(1, "l")

print(5 + a.kwarg)
```
And here, with the type, mypy will error properly at the instantiation of a

Reply:
Okay, understood completely. I'll go ahead and add the types.